### DISCUSSION TITLE
Add an option to disable automatic update checks

### DISCUSSION THREAD

**Discussion by raiguard** [ppy/osu#17749] [Ideas] [answered]
For the average user, having automatic update checks is very good. However, I am on Arch Linux, and I get osu! from the [Arch User Repository](https://aur.archlinux.org/packages/osu-lazer-bin) (it downloads the AppImage from the GitHub release). Thus, the update notification is completely redundant for me - I update osu!lazer when I update my system using pacman.

Could you please add an option to disable automatic update checks for situations like this?

Thank you for your hard work on osu!

**Comment by bdach** [comment, marked as answer]
The AUR package is unofficial and not supported by anyone here. The only official distribution method is the appimage which will eventually automatically update.

**Reply by raiguard** [reply]
Yes, I'm aware that it's not official. I simply provided it as my usecase - the actual request is to have a way to disable automatic update checks. Honestly, even when the appimage eventually does get auto-update support, I'd prefer to disable it and update it with ~~pacman~~ paru anyway.

I think many Linux users (not just Arch) would agree with me that updating using the package manager would be preferential to it updating itself.

**Reply by peppy** [reply]
There's already and envvar flag that such packages can set to change the auto updates to only notify of a new release being available, while not actually download anything.

**Reply by raiguard** [reply]
Sorry, I think I was unclear. I should have titled the question "notifications" instead of "checks". The current behavior is as you describe - it notifies me of new updates, but doesn't download anything. What I'm asking for is the ability to suppress that notification. I update my system every 1-2 weeks, and more often than not, osu! has an update. But I don't need to know about it until I run the command.

I understand if you don't want to add another option just for this. It's not a dealbreaker by any means, but it would be nice. Thanks!

![image](https://user-images.githubusercontent.com/3515394/162596216-472aacf4-c9b2-4628-b8ff-0ec72aeb9cef.png)

**Reply by peppy** [reply]
https://github.com/flathub/flathub/pull/2619#issuecomment-969731305 this is the closest you will get. osu! will notify about updates because there are many times the previous version will not work when we push a new update.

**Reply by raiguard** [reply]
Fair enough, thank you. I appreciate your time.
